tests/unit/s2n_x509_validator_test.c

```diff
@@ -413,8 +429,11 @@ int main(int argc, char **argv)
         s2n_x509_trust_store_wipe(&trust_store);
     };
 
-    /* test expired certificate fails as untrusted*/
-    {
+    /* test expired certificate fails as untrusted. The test fails on platforms
+     * where time_t is 4 bytes because attempting to mock the system time to a
+     * post-2038 date overflows the time_t.
+     */
+    if (sizeof(time_t) != 4) {
```

Comment:
It might be worth wrapping this check in a named function, since you make it in a few places. Could also consolidate some of the comments.